src/bank.rs

```diff
@@ -1061,6 +1092,33 @@ mod tests {
         assert!(!validator_bank.is_leader);
     }
     #[test]
+    fn test_hash_bank_state() {
+        let mint = Mint::new(2_000);
+        let seed = [0u8; 32];
+        let mut rnd = GenKeys::new(seed);
+        let keypairs = rnd.gen_n_keypairs(5);
+        let (ledger0, _pubkey) =
+            create_sample_ledger_with_mint_and_keypairs(mint.clone(), &keypairs);
+        let (ledger1, _pubkey) =
+            create_sample_ledger_with_mint_and_keypairs(mint.clone(), &keypairs);
+
+        let bank0 = Bank::default();
+        bank0.process_ledger(ledger0).unwrap();
+        let bank1 = Bank::default();
+        bank1.process_ledger(ledger1).unwrap();
+
+        assert_eq!(bank0.hash_bank_state(), bank1.hash_bank_state());
+
+        let pubkey = keypairs[0].pubkey();
+        bank0
+            .transfer(1_000, &mint.keypair(), pubkey, mint.last_id())
+            .unwrap();
```

Comment:
I'd like to offer a competing implementation of `hash_bank_state()`:

```rust
fn hash_bank_state(&self) {
    Hash::default()
}
```

Looks like you'll want to assert *not* equal on the original hash here.

Reply:
Lol, yes